### PR TITLE
feat: implement this silly goose

### PR DESCRIPTION
**Pull request by wolfenrain**
## Status

**READY**

## Description

Implementing this silly goose!

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,4 +8,168 @@
...
 
-Just a silly goose.
\ No newline at end of file
+---
+
+Just as geese migrate, you too can migrate your code with the help of Goose.
+
+---
+
+Goose is a minimalistic migration framework that has no opinions on how you should migrate, it only provides a structure to do so.
```
Comment (scarletteliza) [inline README.md:17]:
```suggestion
Goose is a minimal migration framework that has no opinions on how you should migrate, it only provides a structure to do so.
```

File: README.md [hunk 2]
```diff
@@ -8,4 +8,168 @@
...
+
+You can use Goose wherever you want, with whatever you want. May that be [Flutter](https://flutter.dev) with [Isar](https://isar.dev) or [Dart Frog](https://dartfrog.vgv.dev) with [Hive](https://pub.dev/packages/hive) or X with Y, this silly goose can do it all!
```
Comment (scarletteliza) [inline README.md]:
```suggestion
You can use Goose wherever you want, with whatever you want. Whether that's [Flutter](https://flutter.dev) with [Isar](https://isar.dev) or [Dart Frog](https://dartfrog.vgv.dev) with [Hive](https://pub.dev/packages/hive) or X with Y, this silly goose can do it all!
```

File: README.md [hunk 3]
```diff
@@ -8,4 +8,168 @@
...
+  await goose.up();
+
+  // Migrate all the way down through all our migrations.
+  await goose.down();
+}
+```
+
+### Migrating up and down to a certain migration
+
+In the above example we called `goose.up` to migrate all the way up but you can also pass a migration id to migrate up that given migration:
```
Comment (scarletteliza) [inline README.md]:
```suggestion
In the above example we called `goose.up` to migrate all the way up but you can also pass a `migration id` to migrate up that given migration:
```
yah or nah?

File: README.md [hunk 4]
```diff
@@ -8,4 +8,168 @@
...
+      ...
+    ],
+  );
+
+  // This will migrate up to that specific migration.
+  await goose.up(to: 'my_specific_migration');
+}
+```
+
+The same can be done with the `down`, keep in mind that the migration you specific wont be migrated down.
```
Comment (scarletteliza) [inline README.md]:
```suggestion
The same can be done with the `down`, but keep in mind that the migration you specify won't be migrated down.
```

File: README.md [hunk 5]
```diff
@@ -8,4 +8,168 @@
...
+    for (final migration in migrations) {
+      print('${migration.name} has the description "${migration.description}" and is ${migration.isMigrated ? 'migrated' : 'not migrated'}');
+    }
+  })
+
+  await goose.up();
+}
+```
+
+And if you want to retrieve the current state of the migrations:
```
Comment (scarletteliza) [inline README.md:156]:
```suggestion
And if you want to retrieve the current state of the migrations:
```
singular or plural here?

**Review by scarletteliza — COMMENTED** [review]
lgtm 🪿

File: README.md (hunk 5 above)
Comment (wolfenrain) [reply README.md:156]: Plural, nice check tho!

Comment (scarletteliza) [reply README.md:156]: so one state, multiple migrations? if so, good to go

**Pull request merged**